packages/astro/src/types/public/integrations.ts

```diff
@@ -246,12 +242,105 @@ export interface AstroIntegration {
 /**
  * A smaller version of the {@link RouteData} that is used in the integrations.
  */
-export type IntegrationRouteData = Omit<
+export type IntegrationRouteData = Pick<
 	RouteData,
-	'isIndex' | 'fallbackRoutes' | 'redirectRoute'
+	'route' | 'distURL'
 > & {
+	/**
+	 * {@link RouteData.component}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	component: RouteData['component'];
+
+	/**
+	 * {@link RouteData.generate}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	generate: RouteData['generate'];
+
+	/**
+	 * {@link RouteData.params}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	params: RouteData['params'];
+
+	/**
+	 * {@link RouteData.pathname}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	pathname?: RouteData['pathname'];
+
+	/**
+	 * {@link RouteData.pattern}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	pattern: RouteData['pattern'];
+
+	/**
+	 * {@link RouteData.segments}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	segments: RouteData['segments'];
+
+	/**
+	 * {@link RouteData.type}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	type: RouteData['type'];
+
+	/**
+	 * {@link RouteData.prerender}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	prerender: RouteData['prerender'];
+
+	/**
+	 * {@link RouteData.redirect}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
+	 */
+	redirect?: RouteData['redirect'];
+
 	/**
 	 * {@link RouteData.redirectRoute}
+	 * @deprecated Use `routes` from the `astro:routes:resolved` hook
 	 */
 	redirectRoute?: IntegrationRouteData;
 };
+
+export interface IntegrationResolvedRoute
```

Comment:
The discussion was marked as resolved because of commits, but we need to figure out the renaming of some fields